pkg/controller/core/workload_controller.go

```diff
@@ -193,16 +186,26 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 		if updated {
 			return ctrl.Result{}, workload.ApplyAdmissionStatus(ctx, r.client, &wl, true)
 		}
-	} else if !apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadEvicted) {
-		// if job is not active and does not have condition reason of WorkloadEvictedByDeactivation, update its condition
-		workload.SetEvictedCondition(&wl, kueue.WorkloadEvictedByDeactivation, "The workload is deactivated")
-		if err := workload.ApplyAdmissionStatus(ctx, r.client, &wl, true); err != nil {
-			return ctrl.Result{}, fmt.Errorf("setting eviction: %w", err)
+	} else {
+		var updated, evicted bool
+		if wl.Status.RequeueState != nil {
+			wl.Status.RequeueState = nil
+			updated = true
 		}
-		if wl.Status.Admission != nil {
-			metrics.ReportEvictedWorkloads(string(wl.Status.Admission.ClusterQueue), kueue.WorkloadEvictedByDeactivation)
+		if !apimeta.IsStatusConditionTrue(wl.Status.Conditions, kueue.WorkloadEvicted) {
+			workload.SetEvictedCondition(&wl, kueue.WorkloadEvictedByDeactivation, "The workload is deactivated")
```

Comment:
What I don't like about this is that we are losing context as to why the workload was deactivated.

In addition to adding the evicted condition here, could we add it inside `reconcileNotReadyTimeout` so that, if the Kueue controller doesn't restart of fail to send the status update, we keep that information in the Evicted reason and message?

Reply:
IIUC, we have the `status.requeueState` in this function, just before we clean it just above, can we use it in the message? We probably don't want to set "WorkloadInactive" reason prior to setting `active=false`.

I think long-term the cleanest would be if `active` was in status, then we could do the following in one request:
```yaml
active: false
conditions:
  Evicted: true
  Reason: PodsReadyTimeoutExceeded
  Message: The PodsReady timeout of 5min is exceeded
  Deactivated: true
  Reason: PodsReadyBackoffExceeded
  Message: Deactivated the workload after 5 attempts due to exceeding the PodsReady timeout.
```

Reply:
> IIUC, we have the status.requeueState in this function, just before we clean it just above, can we use it in the message, for example to say "The workload is deactivated due to 5 failed attempts to run all pods within the PodsReady timeout"? We probably don't want to set "WorkloadInactive" reason prior to setting active=false.

@mimowo Actually, we have the similar discussion here: https://github.com/kubernetes-sigs/kueue/issues/2175
So, could we work on enhancing the reason as another feature to focus on fixing the bug?

Reply:
SGTM. I suspect you want to have #2175 in 0.7?

Reply:
> I suspect you want to have https://github.com/kubernetes-sigs/kueue/issues/2175 in 0.7?

Yes, I would like to include #2175 in v0.7.

Reply:
> So, could we work on enhancing the reason as another feature to focus on fixing the bug?

sgtm, I was actually suggesting that the current place is ok, since we have the data at hand if we want to build the message. I would be fine to build the message like "The workload is deactivated due to exceeding the PodsReady timeout 5 times in a row.". I'm also ok leave it for https://github.com/kubernetes-sigs/kueue/issues/2175.